src/west/configuration.py

```diff
@@ -37,13 +37,13 @@
 
 import configparser
 import os
-import pathlib
+from pathlib import PureWindowsPath, Path
```

Comment:
Comment to whole commit, not this line in particular.
So nice that Python 3.6 is minimum required.

Reply:
Amen, f-strings and os.PathLike are both great. Also many thanks to @gmarull for contributing a type checking framework that helps convince me we're not doing anything egregiously wrong in this PR.